Tests/MaterialLib/TestPermeabilityModel.cpp

```diff
@@ -0,0 +1,89 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ * \file   TestPermeabilityModel.cpp
+ *
+ * Created on August 17, 2016, 4:00 PM
+ */
+#include <gtest/gtest.h>
+
+#include "TestTools.h"
+
+#include "MaterialLib/PorousMedium/Permeability/createPermeabilityModel.h"
+
+using namespace MaterialLib;
+using namespace MaterialLib::PorousMedium;
+
+// Mock of the gradient of shape functions
+struct MockGradPhi
+{
+    MockGradPhi()
+    {
+        dphi.resize(3, 3);
+        dphi(0, 0) = 1.0;
+        dphi(0, 1) = 1.0;
+        dphi(0, 2) = 1.0;
+        dphi(1, 0) = 2.0;
+        dphi(1, 1) = 2.0;
+        dphi(1, 2) = 2.0;
+        dphi(2, 0) = 3.0;
+        dphi(2, 1) = 3.0;
+        dphi(2, 2) = 3.0;
+
+        trans_dphi = dphi.transpose();
+    }
+
+    CoefMatrix dphi;
+    CoefMatrix trans_dphi;
+};
+
+static MockGradPhi mock_grad_phi;
```

Comment:
Google test's test fixtures can be used rather effectively instead of global objects.
mock_grad_phi must be const or reinitialized in every test (but the test fixtures are preferable).